### ISSUE TITLE
enhancements to #245

### ISSUE THREAD

**Issue by ttodua** [jenkinsci/publish-over-ssh-plugin#327]
### What feature do you want to see added?

On top of https://github.com/jenkinsci/publish-over-ssh-plugin/pull/245 PR, to be addressed these things:

1) "the added feature checkbox is only available from the Admin - > Configure area,   but it not configurable at job level. Perhaps extend to be able to override the Admin setting at job level.

2)  at this moment, the added feature compares md5's of the files to determine which is changed. However, the plugin had an additional option (checkbox or so), so if that's checked, plugin will only download the specific files (which are presented as "changed/added/removed" from github push action, from example payload it's visible which files are changed on the that last  commit:  https://pastebin.com/raw/u8u0B4PB ) . so, at first, the plugin downloaded only those files (instead of whole repo) and then it continue doing what it does (comparing md5 etc...)

**Comment by ttodua** [comment]
I'll add bounty of $100 on this.

**Comment by parvit** [comment]
@ttodua did you receive my email?

**Comment by ttodua** [comment]
hi.
yes I have read, can you tell  specifically what do you want me to explain further?

**Comment by ttodua** [comment]
I think @parvit was about to do this specific bountry, but @parvit do you agree that you can do this?

**Comment by parvit** [comment]
Yes, sorry for the delay, life gets in the way of these things but i'll get to it soon.

**Comment by parvit** [comment]
@ttodua quick question, would you want the job-level config to override the global config or should they be in OR? meaning either active is active for the job.

**Comment by ttodua** [comment]
@parvit, at this moment, we had paused on this too. if you haven't yet started, please halt on this at this moment, till my further notice, if you are not against.
